halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/provider/v1/KubernetesProvider.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.deploy.provider.v1;
+
+import com.netflix.spinnaker.halyard.deploy.component.v1.ComponentService;
+import com.netflix.spinnaker.halyard.deploy.component.v1.ComponentType;
+import com.netflix.spinnaker.halyard.deploy.job.v1.JobRequest;
+import com.netflix.spinnaker.halyard.deploy.job.v1.JobStatus;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import org.springframework.stereotype.Component;
+
+@Component
+public class KubernetesProvider extends Provider {
+  private static final int MIN_POLL_INTERVAL_SECONDS = 1;
```

Comment:
This already being a spring component, you might want to externalize these properties.